### ISSUE TITLE
stack liquid templating makes a man drink: IF branch or ELSE branch doesn't get executed

### ISSUE THREAD

**Issue by matti** [kontena/kontena#1890]
See this one from
```yaml
    # {% if lb_service %}
    ports:
      - 8086:8086
    # {% else %}
    ports:
      - 8086:8086
    # {% endif %}
```

below:
```yaml
stack: matti/influxdb
version: 0.0.1
description: base
expose: api

variables:
  image:
    type: string
    default: influxdb:1.2-alpine
    from:
      prompt:
  affinity:
    type: string
    default: label!=no-influxdb
    from:
      prompt:
  lb_service:
    type: string
    required: false
    from:
      service_link:
        prompt: Pick a loadbalancer
        image: kontena/lb
  lb_domain:
    only_if: lb_service
    type: string
    default: influxdb.example.net
    from:
      prompt:
  lb_https_redirect:
    only_if: lb_service
    type: boolean
    from:
      prompt:

services:
  api:
    image: {{ image }}
    affinity:
      - {{ affinity }}
    stateful: true
    deploy:
      strategy: ha
      wait_for_port: 8086
      interval: 1d
    volumes:
      - /var/lib/influxdb
    environment:
    # {% if lb_service %}
      - KONTENA_LB_INTERNAL_PORT=8086
      - KONTENA_LB_VIRTUAL_HOSTS={{ lb_domain }}
      # {% if lb_https_redirect %}
      - KONTENA_LB_CUSTOM_SETTINGS=redirect scheme https if !{ ssl_fc }
      # {% endif %}
    # {% endif %}

    # {% if lb_service %}
    ports:
      - 1234:8086
    # {% else %}
    ports:
      - 4321:8086
    # {% endif %}

    links:
    # {% if lb_service %}
      - {{ lb_service }}
    # {% endif %}
```

**Comment by matti** [comment]
@kke ?

**Comment by matti** [comment]
So this always results in: `8086/tcp`

```
CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS              PORTS                                         NAMES
78fdfb0f8606        influxdb:1.2-alpine         "/w/w /entrypoint...."   14 seconds ago      Up 13 seconds       8086/tcp                                      influxdb.api-1
```

**Comment by kke** [comment]
Both of them seem to have the port 8086? :)

**Comment by matti** [comment]
... that wasn't the point: they don't get mapped. edited it now.

**Comment by kke** [comment]
Have you tried 
```yaml
ports:
  # {% if lb_service %}
  - 1234:8086
  # {% else %}
  - 4567:8086
  # {% endif %}
```

?

**Comment by SpComb** [comment]
Odd, with `kontena stack validate`, if I select an `lb_service`, then it does render the ports correctly, but selecting `<none>` to omit the optional `lb_service`, it does not render either of the if-else options:

```yaml
stack: matti/influxdb
version: 0.0.1
description: base
expose: api
variables:
  image:
    label: image
    type: string
    default: influxdb:1.2-alpine
    from:
      prompt: 
      env: image
    to:
      env: image
    errors: {}
    value: influxdb:1.2-alpine
  affinity:
    label: affinity
    type: string
    default: label!=no-influxdb
    from:
      prompt: 
      env: affinity
    to:
      env: affinity
    errors: {}
    value: label!=no-influxdb
  lb_service:
    label: lb_service
    type: string
    from:
      service_link:
        prompt: Pick a loadbalancer
        image: kontena/lb
      env: lb_service
    to:
      env: lb_service
    required: false
    errors: {}
    value: 
  lb_domain:
    label: lb_domain
    type: string
    default: influxdb.example.net
    from:
      prompt: 
      env: lb_domain
    to:
      env: lb_domain
    only_if: lb_service
    errors: {}
    value: 
  lb_https_redirect:
    label: lb_https_redirect
    type: boolean
    from:
      prompt: 
      env: lb_https_redirect
    to:
      env: lb_https_redirect
    only_if: lb_service
    errors: {}
    value: 
  STACK:
    label: STACK
    type: string
    from:
      env: STACK
    to:
      env: STACK
    errors: {}
    value: influxdb
  GRID:
    label: GRID
    type: string
    from:
      env: GRID
    to:
      env: GRID
    errors: {}
    value: development
services:
  api:
    image: influxdb:1.2-alpine
    affinity:
    - label!=no-influxdb
    stateful: true
    deploy:
      strategy: ha
      wait_for_port: 8086
      interval: 1d
    volumes:
    - "/var/lib/influxdb"
    environment: 
    links: 
```

**Comment by matti** [comment]
yes. and also verified that this "works" aka port can be mapped:

```
    # {% if lb_service %}
    ports:
      - 8086:8086
    # {% else %}
    ports:
      - 8086:8086
    # {% endif %}

    # this will map
    ports:
      - 8086:8086
```

**Comment by SpComb** [comment]
With  #1884 that case raises an error:

```
> Pick a loadbalancer <none>
> Enter image : influxdb:1.2-alpine
> Enter affinity : label!=no-influxdb
 [error] Liquid error: undefined variable lb_service
```

I assume what's happening is that the optional `lb_service` variable is not defined, and `Liquid::Template.render(.., strict_variables: true)` fails with an error. That error is just ignored in the current CLI, and it continues with the rest of the template, omitting the entire if-else-endif block.

**Comment by matti** [comment]
while at it also fix #1888 for double 🍻  rewards

**Comment by matti** [comment]
same bug here?

```
$ kontena stack install matti/weavescope

> Pick a loadbalancer lb-ingress/lb
> Enter image : weaveworks/scope:0.17.1
> Enter lb_domain : weavescope.lol.com
> Enable lb_https_redirect : **no**
 [done] Creating stack weavescope
 [done] Triggering deployment of stack weavescope
 [done] Waiting for deployment to start
 [done] Deploying service app
 [done] Deploying service probe
```

```
$ kontena service show weavescope/app
[clipped]
  env:
    - KONTENA_LB_INTERNAL_PORT=4040
    - KONTENA_LB_VIRTUAL_HOSTS=weavescope.lol.com
    - **KONTENA_LB_CUSTOM_SETTINGS=redirect scheme https if !{ ssl_fc }**
```

**Comment by SpComb** [comment]
Fixed in #1884 - seems I left the magic out of the commit message
